bin/setup

```diff
@@ -59,7 +61,9 @@ Dir.chdir APP_ROOT do
   run "gem install mailcatcher"
 
   puts "\n== Preparing database =="
+  run 'bin/rake db:environment:set RAILS_ENV=development'
```

Comment:
This doesn't work for me if the `upaya_development` database doesn't already exist, which is why we need the `db:create` command. It looks like the new Rails 5 command for setting the db environment tries to look up the database that's defined in  `database.yml`. You can test this locally by running `dropdb upaya_development` and then `bin/rake db:environment:set RAILS_ENV=development`. It will throw an exception.

I'm not sure why the Rails team made this change. It's so annoying. I don't recall ever having any issues running DB commands against the proper environments in previous Rails versions.

Reply:
I think there might be a bug in Rails 5.1.5. I found this issue: https://github.com/rails/rails/issues/26731

When I run `dropdb upaya_development`, and then `bundle exec rake db:create` or `bin/rails db:create`, with or without `RAILS_ENV=development`, the environment is being set to `test` for some reason. I checked this by running `psql upaya_development` and then `select * from ar_internal_metadata;`, which shows the environment as test.

This bug (https://github.com/rails/rails/issues/32202) was for the opposite, where it was settting the environment for the test DB to `development`, and the comment in that issue says this was fixed in Rails 5.1.5, but maybe it wasn't completely fixed?

Reply:
There is definitely a bug in there somewhere, I just haven't dug in. Hopefully third time is a charm, this works for me when:
- no database exists
- database exists and has environment `test`
- database exists and has environment `development`

I remove the extra `RAILS_ENV=development` because they aren't necessary, I'm guess they were there to try to account for this behavior.